client/src/featureform/register.py

```diff
@@ -2479,6 +2479,45 @@ def df_transformation(self,
         self.__resources.append(decorator)
         return decorator
 
+    def ondemand_feature(self,
+                          tags: List[str],
+                          properties: dict,
+                          variant: str = "default",
+                          name: str = "",
+                          owner: Union[str, UserRegistrar] = "",
+                          description: str = "",
+                          ):
+        """On Demand Feature decorator.
+
```

Comment:
Do we want to add an example section here?

```
    **Examples**
    ```python
    @ondemand_feature
    def avg_user_transactions():
        pass
    ```
```